system_testing.py

```diff
@@ -168,6 +168,41 @@ def build_run_compare(test, tag, branch, local_precice, force_rebuild, rm_all):
             comparison(pathToRef, pathToOutput)
 
 
+def compose_tag(docker_username, base, features, branch):
+    """
+    Compose a tag based on certain features of the image. Our tagging system follows this scheme:
+
+        DOCKER_USER/BASE-FEATURES-BRANCH
+
+    Example:
+        precice/precice-ubuntu1604.home-develop
+        describes an image with the following properties:
+
+    DOCKER_USER is precice.
+        I.e. image will be pushed to https://hub.docker.com/u/precice
+
+    BASE is precice.
+        I.e. the image contains a build of precice.
+
+    FEATURES are ubuntu1604 and home.
+        I.e. ubuntu1604 is used as operating system and preCICE is installed in a user directory
+
+    BRANCH is develop.
+        I.e. image is based on the source code provided at precice:develop, https://github.com/precice/precice/tree/develop.
+    """
+    features_list = []
+    features_list.append(features["os"]) if "os" in features else None
```

Comment:
You can you `features.get("os")` which returns `None` if the key is missing.

Reply:
Thanks for the hint! I added the changes in https://github.com/precice/systemtests/pull/131/commits/ecc0310f5901d0ecbfcfc18993c5e88e187e2c53. Unfortunately for `"petsc"` and `"mpich"` I could not entirely get rid of the `else None` parts.

Reply:
There is a slight problem with using `features.get(key)` instead of `features_list.append(...) if key in features else None`: The first approach appends `None` to the list, if `key` is not found, the second approach does nothing. Therefore, we have to filter away all the `None` elements as a postprocessing step, like introduced in https://github.com/precice/systemtests/pull/131/commits/56ba8af122ad5e00002cfb4e51ffb0d66d98c671.

Reply:
This is not ideal, but I still prefer using `features.get(key)` over the other option, since it is clearer from my point of view.